### ISSUE TITLE
Fix ruby 2.7 

### ISSUE THREAD

**Issue by woidda** [shioyama/mobility#360]
## Context

[Ruby 2.7.0](https://www.ruby-lang.org/en/news/2019/12/25/ruby-2-7-0-released/) was released in December 2019 with diverse improvements.

Amongst other things: 

*"Separation of positional and keyword arguments"*
Automatic conversion of keyword arguments and positional arguments is deprecated, and conversion will be removed in Ruby 3.

## Expected Behavior
No warnings are logged to console and mobility is fit for newer ruby versions.

## Actual Behavior
Warnings
```
/.rvm/gems/ruby-2.7.0/gems/mobility-0.8.9/lib/mobility/attributes.rb:209: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call
/.rvm/gems/ruby-2.7.0/gems/mobility-0.8.9/lib/mobility/plugins/presence.rb:27: warning: The called method `read' is defined here
/.rvm/gems/ruby-2.7.0/gems/mobility-0.8.9/lib/mobility/plugins/fallbacks.rb:142: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call
/.rvm/gems/ruby-2.7.0/gems/mobility-0.8.9/lib/mobility/plugins/cache/translation_cacher.rb:19: warning: The called method `read' is defined here
/.rvm/gems/ruby-2.7.0/gems/mobility-0.8.9/lib/mobility/plugins/cache/translation_cacher.rb:24: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call
/.rvm/gems/ruby-2.7.0/gems/mobility-0.8.9/lib/mobility/plugins/active_record/query.rb:37: warning: The called method `read' is defined here
/.rvm/gems/ruby-2.7.0/gems/mobility-0.8.9/lib/mobility/attributes.rb:209: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call
/.rvm/gems/ruby-2.7.0/gems/mobility-0.8.9/lib/mobility/plugins/presence.rb:27: warning: The called method `read' is defined here
/.rvm/gems/ruby-2.7.0/gems/mobility-0.8.9/lib/mobility/plugins/fallbacks.rb:146: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call
/.rvm/gems/ruby-2.7.0/gems/mobility-0.8.9/lib/mobility/plugins/cache/translation_cacher.rb:19: warning: The called method `read' is defined here
/.rvm/gems/ruby-2.7.0/gems/mobility-0.8.9/lib/mobility/attributes.rb:209: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call
/.rvm/gems/ruby-2.7.0/gems/mobility-0.8.9/lib/mobility/plugins/presence.rb:27: warning: The called method `read' is defined here
/.rvm/gems/ruby-2.7.0/gems/mobility-0.8.9/lib/mobility/attributes.rb:209: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call
/.rvm/gems/ruby-2.7.0/gems/mobility-0.8.9/lib/mobility/plugins/presence.rb:27: warning: The called method `read' is defined here
/.rvm/gems/ruby-2.7.0/gems/mobility-0.8.9/lib/mobility/attributes.rb:209: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call
/.rvm/gems/ruby-2.7.0/gems/mobility-0.8.9/lib/mobility/plugins/presence.rb:27: warning: The called method `read' is defined here
/.rvm/gems/ruby-2.7.0/gems/mobility-0.8.9/lib/mobility/attributes.rb:209: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call
/.rvm/gems/ruby-2.7.0/gems/mobility-0.8.9/lib/mobility/plugins/presence.rb:27: warning: The called method `read' is defined here
/.rvm/gems/ruby-2.7.0/gems/mobility-0.8.9/lib/mobility/plugins/fallbacks.rb:150: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call
/.rvm/gems/ruby-2.7.0/gems/mobility-0.8.9/lib/mobility/plugins/cache/translation_cacher.rb:19: warning: The called method `read' is defined here
/.rvm/gems/ruby-2.7.0/gems/mobility-0.8.9/lib/mobility/attributes.rb:209: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call
/.rvm/gems/ruby-2.7.0/gems/mobility-0.8.9/lib/mobility/plugins/presence.rb:27: warning: The called method `read' is defined here
/.rvm/gems/ruby-2.7.0/gems/mobility-0.8.9/lib/mobility/attributes.rb:209: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call
/.rvm/gems/ruby-2.7.0/gems/mobility-0.8.9/lib/mobility/plugins/presence.rb:27: warning: The called method `read' is defined here
/.rvm/gems/ruby-2.7.0/gems/mobility-0.8.9/lib/mobility/attributes.rb:209: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call
/.rvm/gems/ruby-2.7.0/gems/mobility-0.8.9/lib/mobility/plugins/presence.rb:27: warning: The called method `read' is defined here
```

## Possible Fix
The different spots should be kind of straightforward to fix.

**Comment by shioyama** [comment]
This is handled in #357, which I'll try to look at asap. I'm a little worried about extra object creation with `**options` everywhere...

**Comment by shioyama** [comment]
Update: @jeremyevans has made a change to Ruby which [reduces allocations for keyword argument hashes](https://github.com/ruby/ruby/pull/2945). This should allow using `**options` without allocating extra objects.

I'd like to wait until that change is released to merge this, otherwise we'll forget about it and never get back the original performance we had.

Also note, these warnings may soon go away anyway: https://discuss.rubyonrails.org/t/new-2-7-3-0-keyword-argument-pain-point/74980

**Comment by xxx44yyy** [comment]
up

**Comment by gagalago** [comment]
that is the conclusion https://discuss.rubyonrails.org/t/new-2-7-3-0-keyword-argument-pain-point/74980/47 

So I suppose that as soon as ruby 2.7.2 is released, we can continue to work on this pull request?

**Comment by shioyama** [comment]
This should be fixed on 1.0.0.beta1, can you try it out? You'll need to update your configuration to the new format:

https://github.com/shioyama/mobility/wiki/Introduction-to-Mobility-v1.0

**Comment by shioyama** [comment]
Closing since this is fixed in 1.0.0.beta1 and above. If you see warnings, please upgrade.
